src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -725,21 +719,15 @@ GenTree* Lowering::LowerModPow2(GenTree* node)
     BlockRange().InsertAfter(cns2, falseExpr);
     LowerNode(falseExpr);
 
-    GenTree* const cc = comp->gtNewOperNode(GT_CSNEG_MI, type, trueExpr, falseExpr);
-    cc->gtFlags |= GTF_USE_FLAGS;
+    mod->ChangeOper(GT_CSNEG_MI);
+    mod->gtOp1 = trueExpr;
+    mod->gtOp2 = falseExpr;
+    mod->gtFlags |= GTF_USE_FLAGS;
 
     JITDUMP("Lower: optimize X MOD POW2");
     DISPNODE(mod);
-    JITDUMP("to:\n");
-    DISPNODE(cc);
-
-    BlockRange().InsertBefore(mod, cc);
-    ContainCheckNode(cc);
-    BlockRange().Remove(mod);
-
-    use.ReplaceWith(cc);
 
-    return cc->gtNext;
+    ContainCheckNode(mod);
```

Comment:
Are these changes necessary or should they be done in the new PR too?

Reply:
Sort of? I got rid of the `use`. New PR extends this path anyway with the new instruction.

Reply:
But this is unrelated to the fix, right?
I am fine with leaving it to avoid more churn on this PR.

Reply:
It is unrelated, been trying different ways to shape this function. The next PR is the shape  and impl I'm happy with.